Crashes after pressing "arrow back" a lot of times and does not start

The context: I decided to migrate from newpipe to skytube app. I looked for a way to migrate the subscription list from one app to another and didn't find anything (that's an issue on itself, but not this one). Then I decided to subscribe to all my favourite channels manually.

To do that I searched the channel name, pressed the `...` button and selected "Subscribe" in the menu. And repeated this for every one of my subscriptions. After this I wanted to see my feed, but when I pressed the "back" arrow on top of my screen I returned to the previous search page. And the preivous. And so on. I quickly pressed this "arrow" button a lot of times and the app has crashed. "Not a big deal" I thought and started the app again, but... it didn't start. It didn't start even after I cleared the cache and force stopped the app in the app info dialogue. If I clear the app data, I'll have to subscribe to all these channels again. There goes my migration from NewPipe. I guess I'll have to tolerate it's issues for a while more... sorry for my frustration.

<details>
  <summary>Here's what seems related to the crash</summary>

```
11-14 12:09:34.924  1784 15686 I ActivityTaskManager: START u0 {act=android.intent.action.MAIN cat=[android.intent.category.LAUNCHER] flg=0x10200000 cmp=free.rm.skytube.oss/free.rm.skytube.gui.activities.MainActivity bnds=[642,1603][845,1877]} from uid 10115
11-14 12:09:34.951  1784  1889 E system_server: Invalid ID 0x00000000.
11-14 12:09:34.957   731   731 D Zygote  : Forked child process 4640
11-14 12:09:34.962  1784  1898 I ActivityManager: Start proc 4640:free.rm.skytube.oss/u0a148 for pre-top-activity {free.rm.skytube.oss/free.rm.skytube.gui.activities.MainActivity}
11-14 12:09:35.030  4640  4640 I MultiDex: VM with version 2.1.0 has multidex support
11-14 12:09:35.030  4640  4640 I MultiDex: Installing application
11-14 12:09:35.030  4640  4640 I MultiDex: VM has multidex support, MultiDex support library is disabled.
11-14 12:09:35.084  4640  4665 D UpdatesCheckerTask: Current Version Number: 2.975, last displayed: 2.975, show release notes: false
11-14 12:09:35.084  4640  4665 D UpdatesChecker: OSS version - will not be checking for updates.
11-14 12:09:35.104  4640  4640 I MainActivity: Action is : android.intent.action.MAIN
11-14 12:09:35.104  4640  4640 I MainActivity: initMainFragment called android.intent.action.MAIN
11-14 12:09:35.139  4640  4640 W TabLayout: MODE_SCROLLABLE + GRAVITY_FILL is not supported, GRAVITY_START will be used instead
11-14 12:09:35.226  4640  4640 D MainFragment: MAINFRAGMENT RESUMED 2
11-14 12:09:35.226  4640  4640 I MainActivity: onResumeFragments Intent { act=android.intent.action.MAIN cat=[android.intent.category.LAUNCHER] flg=0x10200000 cmp=free.rm.skytube.oss/free.rm.skytube.gui.activities.MainActivity bnds=[642,1603][845,1877] }
11-14 12:09:35.276  4640  4640 I VideoGridAdapter: SUBSCRIPTIONS_FEED_VIDEOS
11-14 12:09:35.288  4640  4640 I VideoGridAdapter: MOST_POPULAR
11-14 12:09:35.292  4640  4640 I VideoGridAdapter: FEATURED
11-14 12:09:35.398  4640  4640 I VideoGridAdapter: BOOKMARKS_VIDEOS
11-14 12:09:35.405  4640  4640 I VideoGridAdapter: DOWNLOADED_VIDEOS
11-14 12:09:35.427  4640  4662 I AdrenoGLES: QUALCOMM build                   : b7efb54, I285e059637
11-14 12:09:35.427  4640  4662 I AdrenoGLES: Build Date                       : 10/31/19
11-14 12:09:35.427  4640  4662 I AdrenoGLES: OpenGL ES Shader Compiler Version: EV031.27.05.02
11-14 12:09:35.427  4640  4662 I AdrenoGLES: Local Branch                     :
11-14 12:09:35.427  4640  4662 I AdrenoGLES: Remote Branch                    :
11-14 12:09:35.427  4640  4662 I AdrenoGLES: Remote Branch                    :
11-14 12:09:35.427  4640  4662 I AdrenoGLES: Reconstruct Branch               :
11-14 12:09:35.427  4640  4662 I AdrenoGLES: Build Config                     : S L 8.0.12 AArch64
11-14 12:09:35.442  4640  4662 I AdrenoGLES: PFP: 0x005ff112, ME: 0x005ff066
11-14 12:09:35.443  4640  4667 E SubscriptionsDb: Error occurred while extracting channel{Id,Name} from JSON
11-14 12:09:35.443  4640  4667 E SubscriptionsDb: org.json.JSONException: No value for channelId
11-14 12:09:35.443  4640  4667 E SubscriptionsDb: 	at org.json.JSONObject.get(JSONObject.java:399)
11-14 12:09:35.443  4640  4667 E SubscriptionsDb: 	at free.rm.skytube.businessobjects.db.SubscriptionsDb.extractVideos(SubscriptionsDb.java:639)
11-14 12:09:35.443  4640  4667 E SubscriptionsDb: 	at free.rm.skytube.businessobjects.db.SubscriptionsDb.getSubscriptionVideoPage(SubscriptionsDb.java:591)
11-14 12:09:35.443  4640  4667 E SubscriptionsDb: 	at free.rm.skytube.businessobjects.db.SubscriptionsDb.getSubscriptionVideoPage(SubscriptionsDb.java:567)
11-14 12:09:35.443  4640  4667 E SubscriptionsDb: 	at free.rm.skytube.businessobjects.db.Tasks.GetSubscriptionsVideosFromDb.getNextVideos(GetSubscriptionsVideosFromDb.java:47)
11-14 12:09:35.443  4640  4667 E SubscriptionsDb: 	at free.rm.skytube.businessobjects.YouTube.Tasks.GetYouTubeVideosTask.collectUntil(GetYouTubeVideosTask.java:135)
11-14 12:09:35.443  4640  4667 E SubscriptionsDb: 	at free.rm.skytube.businessobjects.YouTube.Tasks.GetYouTubeVideosTask.getNextVideos(GetYouTubeVideosTask.java:125)
11-14 12:09:35.443  4640  4667 E SubscriptionsDb: 	at free.rm.skytube.businessobjects.YouTube.Tasks.GetYouTubeVideosTask.doInBackground(GetYouTubeVideosTask.java:102)
11-14 12:09:35.443  4640  4667 E SubscriptionsDb: 	at free.rm.skytube.businessobjects.YouTube.Tasks.GetYouTubeVideosTask.doInBackground(GetYouTubeVideosTask.java:43)
11-14 12:09:35.443  4640  4667 E SubscriptionsDb: 	at android.os.AsyncTask$3.call(AsyncTask.java:378)
11-14 12:09:35.443  4640  4667 E SubscriptionsDb: 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
11-14 12:09:35.443  4640  4667 E SubscriptionsDb: 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
11-14 12:09:35.443  4640  4667 E SubscriptionsDb: 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
11-14 12:09:35.443  4640  4667 E SubscriptionsDb: 	at java.lang.Thread.run(Thread.java:919)
11-14 12:09:35.445  4640  4667 E SubscriptionsDb: Error occurred while extracting channel{Id,Name} from JSON
11-14 12:09:35.445  4640  4667 E SubscriptionsDb: org.json.JSONException: No value for channelId
11-14 12:09:35.445  4640  4667 E SubscriptionsDb: 	at org.json.JSONObject.get(JSONObject.java:399)
11-14 12:09:35.445  4640  4667 E SubscriptionsDb: 	at free.rm.skytube.businessobjects.db.SubscriptionsDb.extractVideos(SubscriptionsDb.java:639)
11-14 12:09:35.445  4640  4667 E SubscriptionsDb: 	at free.rm.skytube.businessobjects.db.SubscriptionsDb.getSubscriptionVideoPage(SubscriptionsDb.java:591)
11-14 12:09:35.445  4640  4667 E SubscriptionsDb: 	at free.rm.skytube.businessobjects.db.SubscriptionsDb.getSubscriptionVideoPage(SubscriptionsDb.java:567)
11-14 12:09:35.445  4640  4667 E SubscriptionsDb: 	at free.rm.skytube.businessobjects.db.Tasks.GetSubscriptionsVideosFromDb.getNextVideos(GetSubscriptionsVideosFromDb.java:47)
11-14 12:09:35.445  4640  4667 E SubscriptionsDb: 	at free.rm.skytube.businessobjects.YouTube.Tasks.GetYouTubeVideosTask.collectUntil(GetYouTubeVideosTask.java:135)
11-14 12:09:35.445  4640  4667 E SubscriptionsDb: 	at free.rm.skytube.businessobjects.YouTube.Tasks.GetYouTubeVideosTask.getNextVideos(GetYouTubeVideosTask.java:125)
11-14 12:09:35.445  4640  4667 E SubscriptionsDb: 	at free.rm.skytube.businessobjects.YouTube.Tasks.GetYouTubeVideosTask.doInBackground(GetYouTubeVideosTask.java:102)
11-14 12:09:35.445  4640  4667 E SubscriptionsDb: 	at free.rm.skytube.businessobjects.YouTube.Tasks.GetYouTubeVideosTask.doInBackground(GetYouTubeVideosTask.java:43)
11-14 12:09:35.445  4640  4667 E SubscriptionsDb: 	at android.os.AsyncTask$3.call(AsyncTask.java:378)
11-14 12:09:35.445  4640  4667 E SubscriptionsDb: 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
11-14 12:09:35.445  4640  4667 E SubscriptionsDb: 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
11-14 12:09:35.445  4640  4667 E SubscriptionsDb: 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
11-14 12:09:35.445  4640  4667 E SubscriptionsDb: 	at java.lang.Thread.run(Thread.java:919)
11-14 12:09:35.452  2168  2168 D ImageWallpaper: wallpaper visibility changes to: false
11-14 12:09:35.469   814   814 E Layer   : [Surface(name=AppWindowToken{11be69c token=Token{3e0df6e ActivityRecord{bc7c6e9 u0 com.android.launcher3/.lineage.LineageLauncher t1168}}})/@0x107f0fd - animation-leash#0] No local sync point found
11-14 12:09:35.469   814   814 E Layer   : [Surface(name=AppWindowToken{11be69c token=Token{3e0df6e ActivityRecord{bc7c6e9 u0 com.android.launcher3/.lineage.LineageLauncher t1168}}})/@0x107f0fd - animation-leash#0] No local sync point found
11-14 12:09:35.469   814   814 E Layer   : [Surface(name=AppWindowToken{90d7791 token=Token{152edb8 ActivityRecord{c5f581b u0 free.rm.skytube.oss/free.rm.skytube.gui.activities.MainActivity t1211}}})/@0x58ed785 - animation-leash#0] No local sync point found
11-14 12:09:35.471  4640  4667 E AndroidRuntime: FATAL EXCEPTION: AsyncTask #2
11-14 12:09:35.471  4640  4667 E AndroidRuntime: Process: free.rm.skytube.oss, PID: 4640
11-14 12:09:35.471  4640  4667 E AndroidRuntime: java.lang.RuntimeException: An error occurred while executing doInBackground()
11-14 12:09:35.471  4640  4667 E AndroidRuntime: 	at android.os.AsyncTask$4.done(AsyncTask.java:399)
11-14 12:09:35.471  4640  4667 E AndroidRuntime: 	at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:383)
11-14 12:09:35.471  4640  4667 E AndroidRuntime: 	at java.util.concurrent.FutureTask.setException(FutureTask.java:252)
11-14 12:09:35.471  4640  4667 E AndroidRuntime: 	at java.util.concurrent.FutureTask.run(FutureTask.java:271)
11-14 12:09:35.471  4640  4667 E AndroidRuntime: 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
11-14 12:09:35.471  4640  4667 E AndroidRuntime: 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
11-14 12:09:35.471  4640  4667 E AndroidRuntime: 	at java.lang.Thread.run(Thread.java:919)
11-14 12:09:35.471  4640  4667 E AndroidRuntime: Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String free.rm.skytube.businessobjects.YouTube.POJOs.YouTubeChannel.getId()' on a null object reference
11-14 12:09:35.471  4640  4667 E AndroidRuntime: 	at free.rm.skytube.businessobjects.YouTube.VideoBlocker.filterByBlacklistedChannels(VideoBlocker.java:183)
11-14 12:09:35.471  4640  4667 E AndroidRuntime: 	at free.rm.skytube.businessobjects.YouTube.VideoBlocker.filter(VideoBlocker.java:95)
11-14 12:09:35.471  4640  4667 E AndroidRuntime: 	at free.rm.skytube.businessobjects.YouTube.Tasks.GetYouTubeVideosTask.doInBackground(GetYouTubeVideosTask.java:107)
11-14 12:09:35.471  4640  4667 E AndroidRuntime: 	at free.rm.skytube.businessobjects.YouTube.Tasks.GetYouTubeVideosTask.doInBackground(GetYouTubeVideosTask.java:43)
11-14 12:09:35.471  4640  4667 E AndroidRuntime: 	at android.os.AsyncTask$3.call(AsyncTask.java:378)
11-14 12:09:35.471  4640  4667 E AndroidRuntime: 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
11-14 12:09:35.471  4640  4667 E AndroidRuntime: 	... 3 more
11-14 12:09:35.480  1784  4671 I DropBoxManagerService: add tag=data_app_crash isTagEnabled=true flags=0x2
11-14 12:09:35.480  1784  3292 W ActivityTaskManager:   Force finishing activity free.rm.skytube.oss/free.rm.skytube.gui.activities.MainActivity
11-14 12:09:35.489  1784  4671 W DropBoxManagerService: Dropping: data_app_crash (1486 > 0 bytes)
11-14 12:09:35.469   814   814 E Layer   : [Surface(name=AppWindowToken{90d7791 token=Token{152edb8 ActivityRecord{c5f581b u0 free.rm.skytube.oss/free.rm.skytube.gui.activities.MainActivity t1211}}})/@0x58ed785 - animation-leash#0] No local sync point found
11-14 12:09:35.495   576   576 I hwservicemanager: getTransport: Cannot find entry android.hardware.graphics.mapper@3.0::IMapper/default in either framework or device manifest.
11-14 12:09:35.496  4640  4662 W Gralloc3: mapper 3.x is not supported
11-14 12:09:35.505  4640  4667 I Process : Sending signal. PID: 4640 SIG: 9
11-14 12:09:35.529  1784  2073 W InputDispatcher: channel '144f77e free.rm.skytube.oss/free.rm.skytube.gui.activities.MainActivity (server)' ~ Consumer closed input channel or an error occurred.  events=0x9
11-14 12:09:35.529  1784  2073 E InputDispatcher: channel '144f77e free.rm.skytube.oss/free.rm.skytube.gui.activities.MainActivity (server)' ~ Channel is unrecoverably broken and will be disposed!
11-14 12:09:35.532  1784 15686 I WindowManager: WIN DEATH: Window{144f77e u0 free.rm.skytube.oss/free.rm.skytube.gui.activities.MainActivity}
11-14 12:09:35.532  1784 15686 W InputDispatcher: Attempted to unregister already unregistered input channel '144f77e free.rm.skytube.oss/free.rm.skytube.gui.activities.MainActivity (server)'
11-14 12:09:35.532  1784 15675 I ActivityManager: Process free.rm.skytube.oss (pid 4640) has died: fore TOP
11-14 12:09:35.539   731   731 I Zygote  : Process 4640 exited due to signal 9 (Killed)
11-14 12:09:35.545  1784  1899 I libprocessgroup: Successfully killed process cgroup uid 10148 pid 4640 in 12ms
11-14 12:09:35.562  1784  1889 W WindowManager: Failed looking up window callers=com.android.server.wm.WindowManagerService.windowForClientLocked:5318 com.android.server.wm.WindowManagerService.removeWindow:1782 com.android.server.wm.Session.remove:176
11-14 12:09:35.568  1784  1888 W ActivityManager: setHasOverlayUi called on unknown pid: 4640
11-14 12:09:35.593  2168  2168 D ImageWallpaper: wallpaper visibility changes to: true
11-14 12:09:35.984  1784  1888 W ActivityTaskManager: Activity top resumed state loss timeout for ActivityRecord{c5f581b u0 free.rm.skytube.oss/free.rm.skytube.gui.activities.MainActivity t-1 f}
```
</details>

--- a/app/src/main/java/free/rm/skytube/businessobjects/YouTube/GetFeaturedVideos.java
+++ b/app/src/main/java/free/rm/skytube/businessobjects/YouTube/GetFeaturedVideos.java
@@ -104,9 +104,13 @@ public class GetFeaturedVideos extends GetYouTubeVideos {
 
 		if (videoList != null) {
 			for (Video video : videoList) {
-				YouTubeVideo yv = new YouTubeVideo(video);
-				yv.setRetrievalTimestamp(System.currentTimeMillis());
-				youTubeVideoList.add(yv);
+				try {
+					YouTubeVideo yv = new YouTubeVideo(video);
+					yv.setRetrievalTimestamp(System.currentTimeMillis());
+					youTubeVideoList.add(yv);
+				} catch (Exception ex) {
+					Logger.e(this, "Unable to convert video: " + video + ", error:" + ex.getMessage(), ex);
+				}
 			}
 		}
 

--- a/app/src/main/java/free/rm/skytube/businessobjects/YouTube/POJOs/YouTubeVideo.java
+++ b/app/src/main/java/free/rm/skytube/businessobjects/YouTube/POJOs/YouTubeVideo.java
@@ -129,33 +129,34 @@ public class YouTubeVideo extends CardData implements Serializable {
 	/**
 	 * Constructor.
 	 */
-	public YouTubeVideo(Video video) {
+	public YouTubeVideo(Video video) throws IllegalArgumentException {
 		this.id = video.getId();
 
 		VideoSnippet snippet = video.getSnippet();
-		if (snippet != null) {
-			this.title = snippet.getTitle();
+		if (snippet == null) {
+			throw new IllegalArgumentException("Missing snippet in "+video);
+		}
+		this.title = snippet.getTitle();
 
-			this.channel = new YouTubeChannel(snippet.getChannelId(), snippet.getChannelTitle());
-			setPublishDate(snippet.getPublishedAt());
+		this.channel = new YouTubeChannel(snippet.getChannelId(), snippet.getChannelTitle());
+		setPublishDate(snippet.getPublishedAt());
 
-			if (snippet.getThumbnails() != null) {
-				Thumbnail thumbnail = snippet.getThumbnails().getHigh();
-				if (thumbnail != null) {
-					this.thumbnailUrl = thumbnail.getUrl();
-				}
-
-				thumbnail = snippet.getThumbnails().getMaxres();
-				if (thumbnail != null) {
-					this.thumbnailMaxResUrl = thumbnail.getUrl();
-				}
+		if (snippet.getThumbnails() != null) {
+			Thumbnail thumbnail = snippet.getThumbnails().getHigh();
+			if (thumbnail != null) {
+				this.thumbnailUrl = thumbnail.getUrl();
 			}
 
-			this.language = snippet.getDefaultAudioLanguage() != null ? snippet.getDefaultAudioLanguage()
-					: (snippet.getDefaultLanguage());
-
-			this.description = snippet.getDescription();
+			thumbnail = snippet.getThumbnails().getMaxres();
+			if (thumbnail != null) {
+				this.thumbnailMaxResUrl = thumbnail.getUrl();
+			}
 		}
+
+		this.language = snippet.getDefaultAudioLanguage() != null ? snippet.getDefaultAudioLanguage()
+				: (snippet.getDefaultLanguage());
+
+		this.description = snippet.getDescription();
 
 		if (video.getContentDetails() != null) {
 			setDuration(video.getContentDetails().getDuration());

--- a/app/src/main/java/free/rm/skytube/businessobjects/db/SubscriptionsDb.java
+++ b/app/src/main/java/free/rm/skytube/businessobjects/db/SubscriptionsDb.java
@@ -60,8 +60,8 @@ public class SubscriptionsDb extends SQLiteOpenHelperEx {
             SubscriptionsVideosTable.COL_YOUTUBE_VIDEO_ID, SubscriptionsVideosTable.COL_PUBLISH_TS, SubscriptionsVideosTable.TABLE_NAME, SubscriptionsVideosTable.COL_CHANNEL_ID);
     private static final String GET_VIDEO_IDS_BY_CHANNEL = String.format("SELECT %s FROM %s WHERE %s = ?",
             SubscriptionsVideosTable.COL_YOUTUBE_VIDEO_ID, SubscriptionsVideosTable.TABLE_NAME, SubscriptionsVideosTable.COL_CHANNEL_ID);
-    private static final String FIND_EMPTY_RETRIEVAL_TS = String.format("SELECT %s FROM %s WHERE %s IS NULL",
-            SubscriptionsVideosTable.COL_YOUTUBE_VIDEO, SubscriptionsVideosTable.TABLE_NAME, SubscriptionsVideosTable.COL_RETRIEVAL_TS);
+    private static final String FIND_EMPTY_RETRIEVAL_TS = String.format("SELECT %s,%s FROM %s WHERE %s IS NULL",
+			SubscriptionsVideosTable.COL_YOUTUBE_VIDEO_ID, SubscriptionsVideosTable.COL_YOUTUBE_VIDEO, SubscriptionsVideosTable.TABLE_NAME, SubscriptionsVideosTable.COL_RETRIEVAL_TS);
 	private static final String sortChannelsASC = "LOWER(" + SubscriptionsTable.COL_TITLE + ") ASC ";
 
 	private static final String SUBSCRIBED_CHANNEL_INFO = String.format("SELECT %1$s,%2$s,%3$s,%4$s,(select max(%6$s) from %7$s videos where videos.%8$s = subs.%1$s) as latest_video_ts FROM %5$s subs",
@@ -611,14 +611,17 @@ public class SubscriptionsDb extends SQLiteOpenHelperEx {
      */
     private List<YouTubeVideo> extractVideos(Cursor cursor, boolean fullColumnList) {
         List<YouTubeVideo> videos = new ArrayList<>();
+        Set<String> invalidIds = new HashSet<>();
         try {
 
             if (cursor.moveToNext()) {
                 final int jsonIdx = cursor.getColumnIndex(SubscriptionsVideosTable.COL_YOUTUBE_VIDEO);
+                final int idIdx = cursor.getColumnIndex(SubscriptionsVideosTable.COL_YOUTUBE_VIDEO_ID);
                 final int retrievalIdx = fullColumnList ? cursor.getColumnIndex(SubscriptionsVideosTable.COL_RETRIEVAL_TS) : -1;
                 final int publishTsIdx = fullColumnList ? cursor.getColumnIndex(SubscriptionsVideosTable.COL_PUBLISH_TS) : -1;
 
                 do {
+                    final String id = cursor.getString(idIdx);
                     final byte[] blob = cursor.getBlob(jsonIdx);
                     final String videoJson = new String(blob);
 
@@ -643,16 +646,33 @@ public class SubscriptionsDb extends SQLiteOpenHelperEx {
                             Logger.e(this, "Error occurred while extracting channel{Id,Name} from JSON", e);
                         }
                     }
-                    // regenerate the video's PublishDatePretty (e.g. 5 hours ago)
-                    video.forceRefreshPublishDatePretty();
-                    // add the video to the list
-                    videos.add(video);
+                    if (video.getChannel() != null) {
+                        // regenerate the video's PublishDatePretty (e.g. 5 hours ago)
+                        video.forceRefreshPublishDatePretty();
+                        // add the video to the list
+                        videos.add(video);
+                    } else {
+                        invalidIds.add(id);
+                    }
                 } while (cursor.moveToNext());
             }
         } finally {
             cursor.close();
         }
+        if (!invalidIds.isEmpty()) {
+            deleteVideosByIds(invalidIds);
+        }
         return videos;
+    }
+
+    private void deleteVideosByIds(Set<String> ids) {
+        for (String id: ids) {
+            Logger.w(this, "delete video by id: "+ id);
+            int rowsDeleted = getWritableDatabase().delete(SubscriptionsVideosTable.TABLE_NAME,
+                        SubscriptionsVideosTable.COL_YOUTUBE_VIDEO_ID + " = ?",
+                        new String[]{id});
+            Logger.w(this, "result "+rowsDeleted+" deleted");
+        }
     }
 
     // Generic channel caching
@@ -670,11 +690,11 @@ public class SubscriptionsDb extends SQLiteOpenHelperEx {
 			if (cursor.moveToNext()) {
 				String title = cursor.getString(cursor.getColumnIndexOrThrow(LocalChannelTable.COL_TITLE));
 				String description = cursor.getString(cursor.getColumnIndexOrThrow(LocalChannelTable.COL_DESCRIPTION));
-				String thumnbail = cursor.getString(cursor.getColumnIndexOrThrow(LocalChannelTable.COL_THUMBNAIL_NORMAL_URL));
+				String thumbnail = cursor.getString(cursor.getColumnIndexOrThrow(LocalChannelTable.COL_THUMBNAIL_NORMAL_URL));
 				String banner = cursor.getString(cursor.getColumnIndexOrThrow(LocalChannelTable.COL_BANNER_URL));
 				long subscriberCount = cursor.getLong(cursor.getColumnIndexOrThrow(LocalChannelTable.COL_SUBSCRIBER_COUNT));
 				long lastCheckTs = cursor.getLong(cursor.getColumnIndexOrThrow(LocalChannelTable.COL_LAST_CHECK_TS));
-				return new YouTubeChannel(channelId, title, description, thumnbail, banner, subscriberCount, false, -1, lastCheckTs);
+				return new YouTubeChannel(channelId, title, description, thumbnail, banner, subscriberCount, false, -1, lastCheckTs);
 			}
 		}
 		return null;

